tests/functional/test_s3.py

```diff
@@ -1519,30 +1519,67 @@ def setUp(self):
     def get_sent_headers(self):
         return self.http_stubber.requests[0].headers
 
-    def test_content_md5_set(self):
+    def test_trailing_checksum_set(self):
         with self.http_stubber:
             self.client.put_object(Bucket="foo", Key="bar", Body="baz")
-        self.assertIn("content-md5", self.get_sent_headers())
+        sent_headers = self.get_sent_headers()
+        self.assertEqual(sent_headers["Content-Encoding"], b"aws-chunked")
+        self.assertEqual(sent_headers["Transfer-Encoding"], b"chunked")
+        self.assertEqual(
+            sent_headers["X-Amz-Trailer"], b"x-amz-checksum-crc32"
+        )
+        self.assertEqual(sent_headers["X-Amz-Decoded-Content-Length"], b"3")
```

Comment:
Should we be validating the checksum algo header we're adding as well?

Reply:
Updated in the latest commit.